subsys/bluetooth/host/gatt.c

```diff
@@ -5139,6 +5146,10 @@ static int sc_set(const char *name, size_t len_rd, settings_read_cb read_cb,
 		id = BT_ID_DEFAULT;
 	} else {
 		id = strtol(next, NULL, 10);
+		if (!(id < CONFIG_BT_ID_MAX)) {
```

Comment:
Any reason why you didn't do this? (applies multiple places). 

```suggestion
		if (id >= CONFIG_BT_ID_MAX)) {
```